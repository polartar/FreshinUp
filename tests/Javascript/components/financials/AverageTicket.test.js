import { createLocalVue, mount } from '@vue/test-utils'
import Vuetify from 'vuetify'
import Component from '~/components/financials/AverageTicket.vue'

describe('financials/TotalSales', () => {
  // Component instance "under test"
  let localVue
  describe('Snapshots', () => {
    test('required props set', () => {
      localVue = createLocalVue()
      const wrapper = mount(Component, {
        localVue: localVue,
        propsData: {
          averageTicket: 23470
        }
      })
      expect(wrapper.element).toMatchSnapshot()
    })
  })
  describe('Visual', () => {
    beforeEach(() => {
      localVue = createLocalVue()
      localVue.use(Vuetify)
    })
    test('averageTicket displayed with currency format', () => {
      const component = mount(Component, {
        propsData: {
          averageTicket: 23470
        },
        localVue
      })
      expect(component.html()).toContain('$234.70')
    })
  })
})
