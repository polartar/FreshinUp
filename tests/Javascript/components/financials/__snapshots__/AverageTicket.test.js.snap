// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`financials/TotalSales Snapshots required props set 1`] = `
<div
  class="py-2"
>
  <div
    class="text-xs-center my-5"
  >
    <div
      class="display-1 primary--text"
    >
      
      $23,470
    
    </div>
     
    <div
      class="font-weight-bold subheading mt-2"
    >
      
      Per Customer
    
    </div>
  </div>
</div>
`;
